Adds toggle port security on network create

snowjet opened the pull request:
##### SUMMARY
Added a new property, 'port_security_enabled' which is a boolean to
enable or disable port security during network creation. The default
behaviour will enable port security, security group and anti spoofing
will act as before. When the attribute is set to False, security
group and anti spoofing are disabled on the ports created on this
network.

Utilises this commit in the shade library - https://github.com/openstack-infra/shade/commit/5b81959b32c52b37ff1465cc3a8d39c8b3a47dee

Its in master so should be available in shade versions > 1.27.1

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
cloud / openstack / os_network 

##### ADDITIONAL INFORMATION
Example use of the os_network module using the new property port_security_enabled:
```
- os_network:
    cloud: mycloud
    state: present
    name: priv_network
    port_security_enabled: false  
```

snowjet commented:
ready_for_review

horacimacias commented:
any chance of having this done for os_port as well?
see https://github.com/ansible/ansible/issues/24694

snowjet commented:
Hi  @emonty  @Shrews  - this is the other half of the change from - https://review.openstack.org/#/c/550646/ . Is there any feedback regarding this pull request? Just looking for support getting it merged.

rjet commented:
ready_for_review

snowjet commented:
ready_for_review

snowjet commented:
bot_status

tbreeds commented on lib/ansible/modules/cloud/openstack/os_network.py:
```diff
@@ -170,6 +179,7 @@ def main():
     project = module.params.get('project')
+    port_security_enabled = module.params['port_security_enabled']
 
     shade, cloud = openstack_cloud_from_module(module, min_version='1.6.0')
```
You need to raise this to 1.27.1 to match your commit message.

snowjet replied:
Thanks. I missed that. Committed and pushed.

snowjet committed:
Adds toggle port security on network create

Added a new property, 'port_security_enabled' which is a boolean to
enable or disable port_security during network creation. The default
behaviour will enable port security, security group and anti spoofing
will act as before. When the attribute is set to False, security
group and anti spoofing are disabled on the ports created on this
network.

Shade supports this option in versions > 1.27.1

snowjet commented:
ready_for_review

mnaser commented on lib/ansible/modules/cloud/openstack/os_network.py:
```diff
@@ -69,7 +69,16 @@
+           Network will use OpenStack defaults if this option is
+           not utilised.
+     type: bool
+     version_added: "2.7"
```
Can you change this to 2.8?

tbreeds replied:
done.

mnaser commented on lib/ansible/modules/cloud/openstack/os_network.py:
```diff
@@ -69,7 +69,16 @@
+     version_added: "2.7"
+requirements:
+     - "openstacksdk"
+     - "shade >= 1.27.1"
```
Please remove the shade dependency, it is no longer required.

tbreeds replied:
done.

mnaser commented on lib/ansible/modules/cloud/openstack/os_network.py:
```diff
@@ -196,10 +207,12 @@ def main():
                     net = cloud.create_network(name, shared, admin_state_up,
-                                               external, provider, project_id)
+                                               external, provider, project_id,
+                                               port_security_enabled)
```
https://github.com/openstack/openstacksdk/blob/master/openstack/cloud/openstackcloud.py#L3382-L3386

`availability_zone_hints` comes after `project_id` so you need to fix the order, or use a keyword argument.

tbreeds replied:
done.

mnaser commented on lib/ansible/modules/cloud/openstack/os_network.py:
```diff
@@ -196,10 +207,12 @@ def main():
                     net = cloud.create_network(name, shared, admin_state_up,
-                                               external, provider)
+                                               external, provider,
+                                               port_security_enabled)
```
Same here, I suggest just adding `port_security_enabled=port_security_enabled` or something like that.  In this case, `project_id` is pointing to `port_security_enabled` value.

tbreeds replied:
done.

mnaser commented:
needs_revision

mnaser commented:
thanks @tbreeds

mnaser commented:
shipit

mnaser commented:
@tbreeds need one more "maintainer" to hit the "shipit" button and you'll be in :)

odyssey4me commented on lib/ansible/modules/cloud/openstack/os_network.py:
```diff
@@ -155,7 +163,8 @@ def main():
         state=dict(default='present', choices=['absent', 'present']),
-        project=dict(default=None)
+        project=dict(default=None),
+        port_security_enabled=dict(default=None, type='bool')
```
Should this default to `False` instead of `None` to match the default in openstacksdk?

jamescassell replied:
No, it should be `True`

Edit: this option should default to true to be consistent with current behavior and good security practice.

514amir replied:
@tbreeds could you make that small fix so it can be merged please!

cloudnull commented:
shipit

odyssey4me commented:
ship

odyssey4me commented:
shipit
